Makefile

```
@@ -15,4 +15,8 @@ install:
 
 .bin/ory: Makefile
 	bash <(curl https://raw.githubusercontent.com/ory/meta/master/install.sh) -d -b .bin ory v0.1.38
-	touch -a -m .bin/ory
\ No newline at end of file
+	touch -a -m .bin/ory
+
+.PHONY: format
+format:
+	npm i && npm run format
```

npm has the severe deficiency that it doesn't automatically install newer versions of dependencies when you update the versions in `package.json`. We are therefore using make to do that for us:

```suggestion
format: node_modules
	npm exec -- prettier --write .
	
node_modules: package-lock.json
	npm ci
	touch node_modules
```

Translated into English:
- building the `format` target requires building the `node_modules` target first
- building the `node_modules` target requires building the `package-lock.json` target first
- what was built when is determined by the `updated` timestamp on the filesystem

When somebody updates `package-lock.json`, it will have a newer `updated` time than `node_modules`. Make will therefore rebuild `node_modules` by running `npm ci` and then `touch node_modules` to update the timestamp of that folder.

---

Damn, to me `make` always was this obscure "c-era" legacy build tool. But this makes sense and seems to be really well though out. Thanks for the explanation! Appreciate it.

I've updated my PR. Thanks for the pointers.